src/lib/db-types.ts

```diff
@@ -20,26 +20,50 @@ export interface DBDocumentMap {
   [key /* documentKey */: string]: DBDocument;
 }
 
-export type DBDocumentType = "section" | "learningLog";
+export type DBDocumentType = DBUserDocumentType | DBOfferingDocumentType;
+export type DBUserDocumentType = "section" | "learningLog";
+export type DBOfferingDocumentType = "published";
 
-export type DBDocumentMetadata = DBSectionDocumentMetadata | DBLearningLogDocumentMetadata;
-export interface DBDocumentMetadataBase {
+export interface DBDocumentMetadata {
```

Comment:
Here's where I'm distinguishing between user documents and offering documents (see commit comment). There's only one type of offering document now (published docs), but there could be more in the future and it's simpler to follow the same pattern for all document types.